### PR TITLE
Convert MySQL report_card table to utf8mb4 to fix 'Incorrect string value'

### LINKED ISSUE #10691: Make MySQL metadata use `utf8mb4`
With 0.33, it seems like the table joins now requires metadata to be `utf8mb4`, but many older MySQL/MariaDB installations has been setup with various older charsets/collations like `latin1` or `utf8`, which causes errors when saving questions like:
```
ERROR middleware.results-metadata :: Error recording results metadata for query:
java.sql.BatchUpdateException (conn=180) Incorrect string value: ‘\xE2\x86\x92 Fi…’ for column ‘result_metadata’ at row 1
```

There has been a surge in reports in the forum the past couple of weeks.

Ways to check the charset/collation (cannot be executed from Metabase editor):
```SQL
-- Set the database name to the Metabase metadata database - change it to match your db:
USE metabase;
-- Show system variables
SHOW VARIABLES WHERE Variable_name LIKE 'character_set_%' OR Variable_name LIKE 'collation%';
-- Show metadata database/tables/columns
SELECT TABLE_SCHEMA,
       TABLE_NAME,
       CCSA.CHARACTER_SET_NAME AS DEFAULT_CHAR_SET,
       COLUMN_NAME,
       COLUMN_TYPE,
       C.CHARACTER_SET_NAME
  FROM information_schema.TABLES AS T
  JOIN information_schema.COLUMNS AS C USING (TABLE_SCHEMA, TABLE_NAME)
  JOIN information_schema.COLLATION_CHARACTER_SET_APPLICABILITY AS CCSA
       ON (T.TABLE_COLLATION = CCSA.COLLATION_NAME)
 WHERE TABLE_SCHEMA=SCHEMA()
   AND C.DATA_TYPE IN ('enum', 'varchar', 'char', 'text', 'mediumtext', 'longtext')
 ORDER BY TABLE_SCHEMA,
          TABLE_NAME,
          COLUMN_NAME;
```

Could Metabase use `utf8mb4`, when people are migrating? Also setting engine to `innodb`, since some older MySQL installations has default engine as `myisam`.

And/or log a warning on startup for existing installations that are not following the recommendation?

A tiny bit similar to #10365

<details>
<summary>Click here to see the commands for doing the change</summary>

***BE CAREFUL!***
This expects you are using `latin1` currently - change it to match your current charset
And input your credentials here instead of: `XXusernameXX`, `XXpasswordXX` and `XXdatabaseXX`

`mysqldump -uXXusernameXX -pXXpasswordXX -c -e --default-character-set=utf8mb4 --single-transaction --skip-set-charset --add-drop-database -B XXmetabaseXX > dump.sql`

`sed -i.bak -e 's/DEFAULT CHARACTER SET latin1/DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci/' -e 's/DEFAULT CHARSET=latin1/DEFAULT CHARSET=utf8mb4/' dump.sql`

`mysql -uXXusernameXX -pXXpasswordXX < dump.sql`
</details>

### PR DESCRIPTION
**Pull request by camsaul**
Fixes #10691 for people with existing setups. Alternative PR to #11573. Which one gets merged depends on how comfortable I am #11573 will not make peoples' DBs explode

### CODE REVIEW

File: src/metabase/driver/mysql.clj [hunk 1]
```diff
@@ -206,8 +206,10 @@
    :zeroDateTimeBehavior "convertToNull"
    ;; Force UTF-8 encoding of results
    :useUnicode           true
-   :characterEncoding    "UTF8"
-   :characterSetResults  "UTF8"
+   :characterEncoding    "UTF-8"
+   :characterSetResults  "UTF-8"
+   :character_set_server "utf8mb4"
+   :connectionCollation  "utf8mb4_unicode_ci"
```
Comment (camsaul) [inline src/metabase/driver/mysql.clj]: these changes aren't part of this fix but this seems to be the way the MySQL dox suggest you set these params.

**Review by flyingmachine — CHANGES_REQUESTED** [review]
If I'm not mistaken, it looks like other tables suffer from this problem? For example https://github.com/metabase/metabase/issues/9786 shows the problem showing up with this query:

```
Query is: INSERT INTO metabase_fieldvalues ( id, created_at, updated_at, values, human_readable_values, field_id ) VALUES ( 1513, '2018-10-09 18:09:47.375', '2018-11-16 18:23:24.457', '["Constant ↔ "]', , 4656 )
```

**Comment by camsaul** [comment]
@flyingmachine PR #11573 is the same thing but converts the entire database. I'm not sure what the implications of converting an entire database to `utf8mb4` will be. Since it involves converting the byte encoding of every text column in the entire database I want to make sure we're not going to be shipping a database migration that potentially takes days to run. That's why I asked for people to test it on #10691 before proceeding further.

This PR exists in case I decide not to pull the trigger on #11573. It's a subset that I'm comfortable merging that would solve 99% of the problems we're seeing.

**Comment by camsaul** [comment]
Closing with out merge in favor of alternative fix PR #11573

**Pull request closed without merging**